eng/SourceBuild.props

```diff
@@ -1,3 +1,5 @@
+<!-- Whenever altering this or other Source Build files, please include @dotnet/source-build-internal as a reviewer. -->
```

Comment:
Does this work similar to a [code owners file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) or will we need to manually add the reviewer?

Reply:
Unfortunately it does not, manual addition would be required.

Initially wanted to add both the comment and the `CODEOWNERS` entry, but for a team to be added as a code owner it needs to have [write permissions](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file) to the repo and I wanted to verify first if this is something we would need. I should've mentioned that in the description though, my bad, forgot to edit it.

If you are OK with granting write permissions to `dotnet/source-build-internal`, I will add the `CODEOWNERS` entry.

Reply:
I'll defer to @sharwell as I am not maintaining this repo any longer.

Reply:
Reached out to Sam in DMs - agreed to add a `CODEOWNERS` entry + write access for the team tag mentioned